### PR TITLE
Added Email Verification Dialog

### PR DESCRIPTION
**Pull request by GAOChengzhan**
#### Register
image

#### Dialog
image

##### Log Out --> Stay anonymous

### CODE REVIEW

**Comment by GAOChengzhan** [comment]
I think unlike CyWeb the redirectURL of the logout function in network-viewer does need to be changed since the URL needs to specify the network's UUID.
And the original `logout` button in the panel also just calls `keycloakClient.logout()`. 

https://github.com/idekerlab/network-viewer/blob/1b70856f6770bb3f817e97ac2ec44d7c332de76f/src/components/NdexLogin/NdexSignInButton/NdexSignInButton.tsx#L79-L81

File: src/index.tsx
Comment (jingjingbic) [inline src/index.tsx:1]: In line 223 and 223, you need to use the same signinByIdToken function call as you did in Cytoscape-web.

**Comment by GAOChengzhan** [comment]
Okay, I replaced with`signinByIdToken ` function.

File: src/components/FooterPanel/SaveNetworkCXButton.tsx
Comment (jingjingbic) [inline src/components/FooterPanel/SaveNetworkCXButton.tsx:1]: The logic seems to be wrong.
If the app has access key, it should always add it to the url as a parameter. If the authenticated add Id_token as well.

**Pull request merged**